js/utility.js

```diff
@@ -22,4 +22,54 @@ export function unhideMultipleElements(array) {
     for (let i = 0; i < array.length; i++) {
         unhideElement(array[i]);
     }
-}
\ No newline at end of file
+}
+
+const levelPopUp = document.querySelector(".levelPopUp");
+
+function triggerPopUp(startWidth, endWidth, element) {
+    const duration = 300; // ms
+    const startTime = performance.now();
+
+    function animate(currentTime) {
+        const elapsed = currentTime - startTime;
+        const progress = Math.min(elapsed / duration, 1);
+
+        const currentWidth = startWidth + (endWidth - startWidth) * progress;
+        element.style.clipPath = `rect(0px ${currentWidth}% 100% 0px round 0%)`
+
+        if (progress < 1) {
+            requestAnimationFrame(animate);
+        }
+    }
+    requestAnimationFrame(animate);
+}
+
+function clearPopUp(elementClass) {
+    const element = document.querySelector(elementClass);
+    triggerPopUp(100, 0, element);
+
+    setTimeout(() => {
+        hideElement(element);
+        levelPopUp.classList.remove("victoryPop", "levelPop", "defeatPop");
+    }, 400);
+}
+
+function showLevelUpPopUp(title, description, elementClass) { // New function for level-up pop-up with 5 second
```

Comment:
i think you are using it a bit incorrectly. This function always interacts with one element, level pop up, so instead of referencing the element with "elementClass", i think you should reference color class (in this case, ".levelPop").
```suggestion
function showLevelUpPopUp(title, description, elementClass) {
    const popUpTexts = document.getElementsByClassName("popUpText");

    popUpTexts[0].textContent = title;
    popUpTexts[1].textContent = description;

    unhideElement(".levelPopUp");
    levelPopUp.classList.remove("defeatPop", "victoryPop", "levelPop");
    levelPopUp.classList.add(elementClass);

    triggerPopUp(0, 100); //what's the need to specify the element?

    setTimeout(() => {
        clearPopUp(elementClass);
    }, 5000);
 }
```